### PR TITLE
Check in frontend

### PR DESCRIPTION
**Pull request by opxz7148**
### Description

- Allow host to close / open check-in for activity via frontend #161 
- Allow Participant to check-in when activities check-in are allowed #162 

Expected behaviour and acceptance criteria of work in this pull request.

- As a host
    - When activity check-in are close. Allow check-in button should shows.
  
        Screenshot 2567-11-02 at 19 01 08

    - When click on allow check-in, check-in code modal will show with a button to get a check in code
  
        Screenshot 2567-11-02 at 19 13 05
        
    - When click on get check-in code. Check-in modal should show check-in code with close check-in button
    
        Screenshot 2567-11-02 at 19 13 12
        
    - In the detail page Allow check-in button should change to show check-in code

        Screenshot 2567-11-02 at 19 13 21
    
    - When click on close check-in inside check-in code modal modal will close it self and show check-in code will change back to Allow check-in

- As a participant
    - When check-in are close detail page should remain the same
    - When host allow for check in check-in button should appear next to chat
       
        Screenshot 2567-11-02 at 19 22 59

    - When click in check-in check-in modal should appear and prompt for check-in code

        Screenshot 2567-11-02 at 19 23 23

    - If check-in code are wrong error message will show

        Screenshot 2567-11-02 at 19 23 40

    - If check-in code are correct check-in modal will disappear and on top of activity detail panel will show text "You've checked in"

        Screenshot 2567-11-02 at 19 24 59

### Checklist

- [x] I have performed a self-review of my code.
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation, if applicable.
- [x] My code follows the Coding Guidelines.

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other (please specify)

### CODE REVIEW

**Review by OmegaOoh — APPROVED** [review]
Everything look fine, We'll leave that red button there for now

File: frontend/src/views/ActivityDetail.vue [hunk 1]
```diff
@@ -230,7 +397,10 @@ export default {
                 this.hosts = JSON.stringify(response.data.host);
                 this.checkHost();
                 this.checkJoined();
+                this.checkCheckedIn();
+                console.log("I'm Ok");
```
Comment (OmegaOoh) [inline frontend/src/views/ActivityDetail.vue:401]: Just saw this, please remove console.log

**Review by OmegaOoh — APPROVED** [review]
Everything seems fine, I found small mistake (just unremoved log)

**Review by Thanawas-Sirilertsathit — APPROVED** [review]
Approve but I found minor bug.
When you checked-in and then you leave the activity then you rejoin the result is you are back into state not checked-in.

**Comment by Thanawas-Sirilertsathit** [comment]
nothing wrong now. The attend would be deleted after leaving activity.

**Pull request merged**